src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -639,5 +630,62 @@ private ContextualFeatureFilterEvaluator GetContextualFeatureFilter(string filte
 
             return filter;
         }
+
+        private async void PublishTelemetry(EvaluationEvent evaluationEvent)
+        {
+            if (evaluationEvent.FeatureDefinition.EnableTelemetry)
+            {
+                if (!TelemetryPublishers.Any())
+                {
+                    _logger.LogWarning("The feature declaration enabled telemetry but no telemetry publisher was registered.");
+                }
+                else
+                {
+                    foreach (ITelemetryPublisher telemetryPublisher in TelemetryPublishers)
+                    {
+                        await telemetryPublisher.PublishEvent(
+                            evaluationEvent,
+                            CancellationToken.None);
+                    }
+                }
+            }
+        }
+
+        private Variant GetVariantFromVariantDefinition(VariantDefinition variantDefinition)
+        {
+            if (variantDefinition == null)
```

Comment:
`if (variantDefinition == null)`

we usually prefer these checks at the caller. It doesn't make much sense to get the variant of a feature definition if there is no feature definition.

Reply:
To me, it's just duplicated logic that I'd prefer to pull into a method. We could either rename this method to be more accurate or I can make a new method in between that still does the null check but then calls this method. 

Essentially, we need to convert from a VariantDefinition to a Variant. If it's null, the resulting variant should also be null.

Reply:
If a rename can end up with a well encapsulated method sure, but as it stands it seems accepting a null variantDefinition isn't appropriate

Reply:
What about `GetVariantFromDefinitionOrDefault`? And the caller passes null as default.

Reply:
`GetVariantFromDefinitionOrDefault` Still doesn't seem to fit the bill. My expectation in that case would be that a variant definition may not have a way to supply a variant and thus a default value is returned. Not that it accepts a null value. 

If we don't have a variant definition, we don't have any business getting anything from it. Caller should just

```
variantDefinition != null ?
    GetVariant(variantDefinition) :
    null;
```